server/api/src/rsGenQuery.cpp

```diff
@@ -805,54 +794,50 @@ _rsGenQuery( rsComm_t *rsComm, genQueryInp_t *genQueryInp,
-        */
-        rei.uoic = &rsComm->clientUser;
-        rei.uoip = &rsComm->proxyUser;
-
-        status = applyRule( "acAclPolicy", NULL, &rei, NO_SAVE_REI );
-        if ( status == 0 ) {
-
-            ruleExecuted = 1; /* No need to retry next time since it
-                             succeeded.  Since this is called at
-                             startup, the Rule Engine may not be
-                             initialized yet, in which case the
-                             default setting is fine and we should
-                             retry next time. */
+    if (ruleExecuted == 0) {
+        std::string svc_role;
+        irods::error ret = get_catalog_service_role(svc_role);
+        int status{};
+        if (!ret.ok()) {
+            irods::log(PASS(ret));
+            // NOLINTNEXTLINE(bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions)
+            status = ret.code();
         }
-    }
 
-    // =-=-=-=-=-=-=-
-    // verify that we are running a query for another agent connection
-    const bool agent_conn_flag = irods::server_property_exists(irods::AGENT_CONN_KW);
+        if (irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role) {
+            constexpr auto priv{0};
+            constexpr auto control_flag{2};
 
-    // =-=-=-=-=-=-=-
-    // detect if a request for disable of strict acls is made
-    int acl_val = -1;
-    char* dis_kw = getValByKey( &genQueryInp->condInput, DISABLE_STRICT_ACL_KW );
-    if ( agent_conn_flag && dis_kw ) {
-        acl_val = 0;
+            // Call directly to avoid additional policy
+            chlGenQueryAccessControlSetup(nullptr, nullptr, nullptr, priv, control_flag);
+        }
+
+        // set a strict acl prop
+        try {
+            irods::set_server_property<std::string>(irods::STRICT_ACL_KW, "on");
+        }
+        catch (irods::exception& e) {
+            irods::log(e);
+            // NOLINTNEXTLINE(bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions)
+            status = e.code();
+        }
+        if (status == 0) {
+            ruleExecuted = 1; /* No need to retry next time since it
+                                   succeeded.  Since this is called at
+                                   startup, the Rule Engine may not be
+                                   initialized yet, in which case the
+                                   default setting is fine and we should
+                                   retry next time. */
+        }
     }
 
     // =-=-=-=-=-=-=-
     // cache the old acl value for reuse later if necessary
-    int old_acl_val =  chlGenQueryAccessControlSetup(
-                           rsComm->clientUser.userName,
-                           rsComm->clientUser.rodsZone,
-                           rsComm->clientAddr,
-                           rsComm->clientUser.authInfo.authFlag,
-                           acl_val );
+    int old_acl_val = chlGenQueryAccessControlSetup(rsComm->clientUser.userName,
```

Comment:
Codacy says... `old_acl_val` is never used (which is true).

Reply:
Done.